src/module/item/spell/index.ts

```diff
@@ -231,16 +233,21 @@ class SpellPF2e extends ItemPF2e {
      * This handles heightening as well as alternative cast modes of spells.
      * If there's nothing to apply, returns null.
      */
-    loadVariant(castLevel: number): SpellPF2e | null {
+    loadVariant(castLevel: number, overlayData: Partial<SpellSource> = {}): SpellPF2e | null {
```

Comment:
The goal behind this function is that it should be able to *resolve* what variant you're trying to load. Instead it should give it a variant id or list of variant ids and then it should fetch it from there.